Tests/AsyncQueueTests/AsyncQueueTests.swift

```diff
@@ -26,8 +26,149 @@ import XCTest
 
 final class AsyncQueueTests: XCTestCase {
```

Comment:
As you read through the below tests: please be thinking of cases I haven't thought of!